src/qs_dispersion_d4.F

```diff
@@ -0,0 +1,170 @@
+!   CP2K: A general program to perform molecular dynamics simulations                              !
+!   Copyright 2000-2024 CP2K developers group <https://cp2k.org>                                   !
+!                                                                                                  !
+!   SPDX-License-Identifier: GPL-2.0-or-later                                                      !
+!--------------------------------------------------------------------------------------------------!
+
+! **************************************************************************************************
+!> \brief Calculation of dispersion using pair potentials
+!> \author JGH
+! **************************************************************************************************
+MODULE qs_dispersion_d4
+
+   USE atomic_kind_types,               ONLY: get_atomic_kind
+   USE cell_types,                      ONLY: cell_type,&
+                                              get_cell
+   USE dftd4,                           ONLY: d4_model,&
+                                              damping_param,&
+                                              get_dispersion,&
+                                              get_rational_damping,&
+                                              new,&
+                                              new_d4_model,&
+                                              realspace_cutoff,&
+                                              structure_type
+   USE kinds,                           ONLY: dp
+   USE particle_types,                  ONLY: particle_type
+   USE qs_dispersion_types,             ONLY: qs_dispersion_type
+   USE qs_force_types,                  ONLY: qs_force_type
+#include "./base/base_uses.f90"
+
+   IMPLICIT NONE
+
+   PRIVATE
+
+   CHARACTER(len=*), PARAMETER, PRIVATE :: moduleN = 'qs_dispersion_d4'
+
+   PUBLIC :: calculate_dispersion_d4_pairpot
+
+! **************************************************************************************************
+
+CONTAINS
+
+#if defined(__DFTD4)
+   ! **************************************************************************************************
+!> \brief ...
+!> \param dispersion_env ...
+!> \param particle_set ...
+!> \param cell ...
+!> \param energy ...
+!> \param force ...
+!> \param atom_of_kind ...
+!> \param virial ...
+! **************************************************************************************************
+   SUBROUTINE calculate_dispersion_d4_pairpot(dispersion_env, particle_set, cell, energy, force, atom_of_kind, virial)
+
+      class(damping_param), ALLOCATABLE                  :: param
+      TYPE(d4_model)                                     :: disp
+      TYPE(structure_type)                               :: mol
+      TYPE(realspace_cutoff)                             :: cutoff
+
+      TYPE(qs_dispersion_type), POINTER, INTENT(IN)                       :: dispersion_env
```

Comment:
Move the block with the declarations of dummy variables upwards. That's how it is the case in all other parts of the code. Don't forget to run `make pretty` again.